api/v4/common_types.go

```diff
@@ -109,6 +109,9 @@ type Spec struct {
 
 	// ServiceTemplate is a template used to create Kubernetes services
 	ServiceTemplate corev1.Service `json:"serviceTemplate"`
+
+	// TopologySpreadConstraint
```

Comment:
Maybe we can add a [link](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/) to the kubernetes documentation here?

Reply:
added